lib/routes/wikipedia/current-events.ts

```diff
@@ -0,0 +1,260 @@
+    }
+    if (!html.endsWith('</p>')) {
+        html = html + '</p>';
+    }
+
+    // Clean up multiple paragraph tags and empty paragraphs
+    html = html.replaceAll(/<\/p>\s*<p>/g, '</p>\n<p>');
+    html = html.replaceAll(/<p>\s*<\/p>/g, '');
+    html = html.replaceAll(/<p>\s*<ul>/g, '<ul>');
+    html = html.replaceAll(/<\/ul>\s*<\/p>/g, '</ul>');
+
+    return html;
+}
+
+// Wiki markup to HTML converter with proper list handling
+function wikiToHtml(wikitext: string): string {
+    let html = wikitext;
+
+    // Apply transformations in order
+    html = convertWikiLinks(html);
+    html = convertExternalLinks(html);
+    html = convertTextFormatting(html);
+    html = processLines(html);
+    html = wrapListItems(html);
+    html = processNestedLists(html);
+    html = wrapInParagraphsAndCleanup(html);
+
+    return html;
+}
+
+function getCurrentEventsDatePath(date: Date): string {
+    const months = ['January', 'February', 'March', 'April', 'May', 'June', 'July', 'August', 'September', 'October', 'November', 'December'];
+
+    const year = date.getFullYear();
+    const month = months[date.getMonth()];
+    const day = date.getDate(); // Not zero-padded
+
+    return `Portal:Current_events/${year}_${month}_${day}`;
+}
+
+async function fetchWikiRaw(pageName: string): Promise<string> {
+    const url = `https://en.wikipedia.org/wiki/${pageName}`;
+    const response = await got(url, {
+        searchParams: {
+            action: 'raw',
+        },
+        headers: {
+            'User-Agent': 'RSSHub/1.0 (+https://github.com/DIYgod/RSSHub)',
+        },
+    });
+
+    return response.body;
+}
+
+export const route: Route = {
+    path: '/current-events',
+    categories: ['new-media'],
+    example: '/wikipedia/current-events',
+    parameters: {},
+    features: {
+        requireConfig: false,
+        requirePuppeteer: false,
+        antiCrawler: false,
+        supportBT: false,
+        supportPodcast: false,
+        supportScihub: false,
+    },
+    radar: [
+        {
+            source: ['en.wikipedia.org/wiki/Portal:Current_events'],
+            target: '/wikipedia/current-events',
+        },
+    ],
+    name: 'Current Events',
+    maintainers: ['aavanian'],
+    handler,
+    description: 'Wikipedia Portal: Current events - Latest news and events from the past 7 days',
+};
+
+async function handler() {
+    // Create array of dates for the past 7 days
+    const dates = Array.from({ length: 7 }, (_, i) => {
+        const date = new Date();
+        date.setDate(date.getDate() - (i + 1));
+        return date;
+    });
+
+    // Fetch all pages in parallel to avoid await-in-loop
+    const fetchPromises = dates.map(async (date) => {
+        const pageName = getCurrentEventsDatePath(date);
+        const cacheKey = `wikipedia:current-events:${pageName}`;
+
+        try {
+            const wikitext = await cache.tryGet(cacheKey, async () => await fetchWikiRaw(pageName), 60 * 60 * 6); // Cache for 6 hours
```

Comment:
Do not hard code the cache duration. It should be managed by https://github.com/DIYgod/RSSHub/blob/380874aca8abfff9b1aad2cb77b50ecccbc77be5/lib/config.ts#L501